exchanges/graphcache/src/store/data.ts

```diff
@@ -156,10 +161,22 @@ export const noopDataState = (
   layerKey: number | null,
   isOptimistic?: boolean
 ) => {
-  initDataState(data, layerKey, isOptimistic);
+  initDataState(OperationType.Read, data, layerKey, isOptimistic);
   clearDataState();
 };
 
+export const getCurrentOperation = (): OperationType => {
+  invariant(
+    currentOperation !== null,
+    'Invalid Cache call: The cache may only be accessed or mutated during' +
+      'operations like write or query, or as part of its resolvers, updaters, ' +
+      'or optimistic configs.',
+    2
```

Comment:
We'll need to add an entry to `docs/graphcache/errors.md` for this

Reply:
I think this is a duplication of an existing error so it’s fine 👍